### PR TITLE
Fix loading model from jar on Java 11

### PR DESCRIPTION
**Pull request by rculbertson**
When loading a model from a jar file, Zoltar's `copyDir` method copies the model to a
temporary directory. `copyDir` calculates the path of the file being copied relative
to the source root directory. It does this using `String.substring`, and assumes the
source root directory has a trailing slash. In Java 11, `ZipPath` (which represents
Paths in jar files) removes the trailing slash. This throws off substring selection,
and causes Zoltar to try to copy files to `/`, which fails with `AccessDeniedException`.

This PR replaces the substring operation with `Path.relativize` which works correctly
whether the trailing slash is present or not.

### CODE REVIEW

File: zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java [hunk 1]
```diff
@@ -98,19 +98,21 @@ public static URI downloadIfNonLocal(final URI path) throws IOException {
...
-      final Path fullDst = Paths.get(dest.toUri().resolve(relative));
+    for (final Path path : paths) {
+      final Path relative = src.relativize(path);
+      // The 'resolve' method can be passed a String or a Path - we must pass String. If copying
+      // from a jar file, the relative path will be ZipPath, while our destination directory will
+      // be UnixPath. This difference will cause resolve to throw ProviderMismatchException, so
+      // we must first convert relative to String.
+      final Path fullDst = dest.resolve(relative.toString());
       final CopyOption[] flags =
-          overwrite ? new CopyOption[] {StandardCopyOption.REPLACE_EXISTING} : new CopyOption[] {};
```
Comment (regadas) [inline zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java]: @rculbertson can you run spotless? `mvn spotless:apply` ... not sure why build didn't fail on travis ... I need to have a look at it

Comment (rculbertson) [reply zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java]: Fixed.

File: zoltar-tests/src/test/java/com/spotify/zoltar/fs/FileSystemExtrasTest.java [hunk 2]
```diff
@@ -94,26 +90,38 @@ public void noCopyIfDefaultFileSystem() throws IOException {
...
   @Test
   public void copyDirectory() throws IOException, URISyntaxException {
     final URI resource = getClass().getResource("/trained_model").toURI();
-    final String abspath = new File(resource).getAbsolutePath();
-    final Path model = Paths.get(abspath);
+    final Path src = new File(resource).toPath();
+    copyDirectory(src);
+  }
 
+  private void copyDirectory(final Path src) throws IOException {
```
[the remark this answers is not in the document]
Comment (rculbertson) [reply zoltar-tests/src/test/java/com/spotify/zoltar/fs/FileSystemExtrasTest.java]: Sure, just pushed that change.

File: zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java [hunk 3]
```diff
@@ -98,19 +98,19 @@ public static URI downloadIfNonLocal(final URI path) throws IOException {
...
-            .filter(path -> !path.equals(src))
-            .map(Path::toUri)
-            .collect(Collectors.toList());
+    final List<Path> paths =
+        Files.walk(src).filter(path -> !path.equals(src)).collect(Collectors.toList());
 
-    for (final URI uri : uris) {
-      final String relative =
-          uri.toString().substring(src.toUri().toString().length(), uri.toString().length());
-      final Path fullDst = Paths.get(dest.toUri().resolve(relative));
```
[the remark this answers is not in the document]
Comment (rculbertson) [reply zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java]: :+1:

**Pull request merged**